tests/func/test_data_status.py

```diff
@@ -423,6 +423,21 @@ def test_missing_remote_cache(M, tmp_dir, dvc, scm, local_remote):
     }
 
 
+def test_not_in_remote_respects_not_pushable(
+    tmp_dir: TmpDir, dvc: Repo, scm, mocker, local_remote
+):
+    stage = dvc.stage.create(outs=["foo"], single_stage=True)
+    stage.outs[0].can_push = False
+    stage.dump()
+
+    tmp_dir.dvc_gen({"dir": {"foobar": "foobar"}})
+    dvc.push()
```

Comment:
Generate
- 1 non-pushable file
- 1 pushable file
- 1 pushable directory
to ensure we test all aspects of `_get_not_in_remote`.